packages/subcommands/src/lib/Subcommand.ts

```diff
@@ -0,0 +1,219 @@
+import { fromAsync, isErr, type Args, Command, type PieceContext, type ChatInputCommand, UserError, type MessageCommand } from '@sapphire/framework';
+import type { Message } from 'discord.js';
+import {
+	ChatInputSubcommandMappings,
+	ChatInputSubcommandGroupMappings,
+	ChatInputSubcommandMappingValue,
+	ChatInputSubcommandToProperty,
+	MessageSubcommandMappings,
+	MessageSubcommandMappingValue,
+	MessageSubcommandToProperty,
+	type SubcommandMappingsArray,
+	MessageSubcommandGroupMappings
+} from './SubcommandMappings';
+import {
+	type ChatInputSubcommandAcceptedPayload,
+	SubcommandPluginEvents,
+	type MessageSubcommandAcceptedPayload,
+	SubcommandPluginIdentifiers
+} from './types/Events';
+
+export class SubCommandPluginCommand<
+	PreParseReturn extends Args = Args,
+	O extends SubCommandPluginCommand.Options = SubCommandPluginCommand.Options
+> extends Command<PreParseReturn, O> {
+	private subcommandsInternalMapping: SubcommandMappingsArray;
```

Comment:
Please make this public instead, so we can access it in error handling events so we can nicely report wrong subcommands for users